pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -236,17 +263,75 @@ func (r *reconcilerBase) createOrPatchDeployment(ctx context.Context, declared *
 		r.log.Error(err, "Managed object update failed, deleting and re-creating",
 			logFieldObject, dRef.String(),
 			logFieldKind, kind)
-		if err := r.client.Delete(ctx, declared); err != nil {
+		if err := deploymentResourceInterface.Delete(ctx, dRef.Name, metav1.DeleteOptions{}); err != nil {
 			return controllerutil.OperationResultNone, err
 		}
-		if err := r.client.Create(ctx, declared); err != nil {
+		data, err := json.Marshal(declared)
+		if err != nil {
+			return controllerutil.OperationResultNone, fmt.Errorf("failed to marshal declared deployment object to byte array: %w", err)
+		}
+		_, err = deploymentResourceInterface.Patch(ctx, dRef.Name, types.ApplyPatchType, data, metav1.PatchOptions{FieldManager: reconcilermanager.ManagerName, Force: &forcePatch})
+		if err != nil {
 			return controllerutil.OperationResultNone, err
 		}
 	}
-
+	if deploymentAfterPatch.GetGeneration() == unObjCurrent.GetGeneration() {
+		return controllerutil.OperationResultNone, nil
+	}
 	return controllerutil.OperationResultUpdated, nil
 }
 
+// preprocessDeployment delete all the fields in allowlist from unstructured object and convert the unstructured object to Deployment object
+func preprocessDeployment(allowList []string, unstructuredDeployment *unstructured.Unstructured) (*appsv1.Deployment, error) {
```

Comment:
Using "process" and "pre-process" in function names is really generic. All functions process things... Can they be named more clearly to say what they do? If it's not clear what the name should be, maybe they could be refactored to improve readability by having functions each do "one thing".